### PR TITLE
[SC-6277]: enable overriding of command args for gofer and spire

### PR DESCRIPTION
**Pull request by vniche**
| Q                        | A
| ------------------------ | ---
| Service                  | (gofer, spire)
| Fixed Issues?            | N/A
| Patch: Bug Fix?          | N/A
| Major: Breaking Change?  | N/A
| Minor: New Feature?      | N/A
| Tests Added + Pass?      | Yes
| Documentation PR         | `[skip-ci]`
| Any Dependency Changes?  | N/A
| License                  | AGPL

### Description of pull request:

Such changes do not break the current implementation. With these changes, container command args can be defined in two ways for gofer:

```yaml
# either
args:
  - agent
  - -v
  - debug

# or
goferMode: agent
logLevel: debug
```

And it goes similar for spire:

```yaml
# either
stream: {}

# or
args:
  - stream
  - -v
  - debug
```

### CODE REVIEW

File: charts/gofer/Chart.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.2.1
+version: 0.2.2
```
Comment (teghnet) [inline charts/gofer/Chart.yaml:24]: I think this constitutes a MINOR bump.

File: charts/gofer/templates/deployment.yaml [hunk 2]
```diff
@@ -39,7 +39,17 @@ spec:
           {{- if .Values.configHcl }}
           args: [ "-c", "/config/gofer.hcl", "{{ .Values.goferMode }}", "-v", "{{ .Values.logLevel | default "debug" }}"]
           {{- else }}
-          args: ["{{ .Values.goferMode }}", "-v", "{{ .Values.logLevel | default "debug" }}"]
+          args:
```
Comment (teghnet) [inline charts/gofer/templates/deployment.yaml:42]: Maybe we could keep consistency between https://github.com/chronicleprotocol/charts/blob/main/charts/spectre/templates/deployment.yaml#L43

and I would not mind having [this](https://github.com/chronicleprotocol/charts/blob/main/charts/spectre/templates/deployment.yaml#L36) also added

File: charts/spire/Chart.yaml [hunk 3]
```diff
@@ -21,7 +21,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.2.1
+version: 0.2.2
```
Comment (teghnet) [inline charts/spire/Chart.yaml:24]: same as above

**Review by teghnet — CHANGES_REQUESTED** [review]

File: charts/gofer/templates/deployment.yaml (hunk 2 above)
Comment (vniche) [reply charts/gofer/templates/deployment.yaml:48]: I've used the spectre patter on bot now, check it out

**Review by teghnet — APPROVED** [review]

**Pull request merged**